src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/domain/entity/update/ReferralUpdate.kt

```diff
@@ -0,0 +1,8 @@
+package uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.entity.update
+
+data class ReferralUpdate(
```

Comment:
Out of interest, is having these `Update`-style classes a common pattern? I know @anandelangovan1 had a couple of comments on this, and how we might want to use the `Referral` entity itself, but I may have misunderstood.

Reply:
Mainly it's for flexibility. Creation might require certain fields to be mandatory, while updates might not. By having separate data classes for each of those functions, we can define different sets of required fields. Conversely, there might be fields that should not be updated once they're set. With separate classes, we're able to just omit those fields from the Update DTO, reducing our overheads. That said, I also like them for their relative simplicity, because they force us to be more explicit with the data we're passing to the API. It's much easier to see what a Create or an Update is actually doing when there are literally fewer fields you're allowed to pass in, and it stops us from needlessly passing in fields which don't need to be updated.